partiql-ast/src/main/java/org/partiql/ast/v1/AstVisitor.java

```diff
@@ -0,0 +1,240 @@
+
+    R visitPathStep(PathStep node, C ctx);
+
+    R visitPathStepField(PathStep.Field node, C ctx);
+
+    R visitPathStepElement(PathStep.Element node, C ctx);
+
+    R visitPathStepAllElements(PathStep.AllElements node, C ctx);
+
+    R visitPathStepAllFields(PathStep.AllFields node, C ctx);
+
+    R visitExprCall(ExprCall node, C ctx);
+
+    R visitExprParameter(ExprParameter node, C ctx);
+
+    R visitExprOperator(ExprOperator node, C ctx);
+
+    R visitExprNot(ExprNot node, C ctx);
+
+    R visitExprAnd(ExprAnd node, C ctx);
+
+    R visitExprOr(ExprOr node, C ctx);
+
+    R visitExprValues(ExprValues node, C ctx);
+
+    R visitExprValuesRow(ExprValues.Row node, C ctx);
+
+    R visitExprArray(ExprArray node, C ctx);
+
+    R visitExprBag(ExprBag node, C ctx);
+
+    R visitExprStruct(ExprStruct node, C ctx);
+
+    R visitExprStructField(ExprStruct.Field node, C ctx);
+
+    R visitExprLike(ExprLike node, C ctx);
+
+    R visitExprBetween(ExprBetween node, C ctx);
+
+    R visitExprInCollection(ExprInCollection node, C ctx);
+
+    R visitExprIsType(ExprIsType node, C ctx);
+
+    R visitExprCase(ExprCase node, C ctx);
+
+    R visitExprCaseBranch(ExprCase.Branch node, C ctx);
+
+    R visitExprCoalesce(ExprCoalesce node, C ctx);
+
+    R visitExprNullIf(ExprNullIf node, C ctx);
+
+    R visitExprSubstring(ExprSubstring node, C ctx);
+
+    R visitExprPosition(ExprPosition node, C ctx);
+
+    R visitExprTrim(ExprTrim node, C ctx);
+
+    R visitExprOverlay(ExprOverlay node, C ctx);
+
+    R visitExprExtract(ExprExtract node, C ctx);
+
+    R visitExprCast(ExprCast node, C ctx);
+
+    R visitExprQuerySet(ExprQuerySet node, C ctx);
+
+    R visitExprMatch(ExprMatch node, C ctx);
+
+    R visitExprWindow(ExprWindow node, C ctx);
+
+    R visitExprWindowOver(ExprWindow.Over node, C ctx);
+
+    R visitQueryBody(QueryBody node, C ctx);
+
+    R visitQueryBodySFW(QueryBody.SFW node, C ctx);
+
+    R visitQueryBodySetOp(QueryBody.SetOp node, C ctx);
+
+    R visitSelect(Select node, C ctx);
+
+    R visitSelectStar(SelectStar node, C ctx);
+
+    R visitSelectProject(SelectProject node, C ctx);
+
+    R visitProjectItem(ProjectItem node, C ctx);
+
+    R visitProjectItemAll(ProjectItem.All node, C ctx);
+
+    R visitProjectItemExpression(ProjectItem.Expression node, C ctx);
+
+    R visitSelectPivot(SelectPivot node, C ctx);
+
+    R visitSelectValue(SelectValue node, C ctx);
+
+    R visitExclude(Exclude node, C ctx);
+
+    R visitExcludePath(ExcludePath node, C ctx);
+
+    R visitExcludeStep(ExcludeStep node, C ctx);
+
+    R visitExcludeStepStructField(ExcludeStep.StructField node, C ctx);
+
+    R visitExcludeStepCollIndex(ExcludeStep.CollIndex node, C ctx);
+
+    R visitExcludeStepStructWildcard(ExcludeStep.StructWildcard node, C ctx);
+
+    R visitExcludeStepCollWildcard(ExcludeStep.CollWildcard node, C ctx);
+
+    R visitFrom(From node, C ctx);
+
+    R visitFromValue(FromValue node, C ctx);
```

Comment:
nit: FromExpr

I am to blame that this is not FromExpr, but FromValue does not _really_ make sense .. so just a nit.